ql/src/java/org/apache/datasketches/hive/hll/DataToSketchUDAF2.java

```diff
@@ -0,0 +1,218 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.datasketches.hive.hll;
+
+import java.util.Arrays;
+
+import org.apache.datasketches.hll.TgtHllType;
+import org.apache.hadoop.hive.ql.exec.Description;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentException;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentTypeException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.udf.generic.AbstractGenericUDAFResolver;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDAFEvaluator;
+import org.apache.hadoop.hive.ql.udf.generic.GenericUDAFParameterInfo;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorUtils;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector.PrimitiveCategory;
+import org.apache.hadoop.hive.serde2.objectinspector.StandardStructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.StructObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorUtils;
+
+/**
+ * Hive UDAF to create an HllSketch from raw data.
+ */
+@Description(
+    name = "dataToSketch",
+    value = "_FUNC_(expr, lgK, type) - "
+        + "Compute a sketch on data 'expr' with given parameters lgK and target type",
+    extended = "Example:\n"
+    + "> SELECT dataToSketch(val, 12) FROM src;\n"
+    + "The return value is a binary blob that can be operated on by other sketch related functions."
+    + " The lgK parameter controls the sketch size and rlative error expected from the sketch."
+    + " It is optional an must be from 4 to 21. The default is 12, which is expected to yield errors"
+    + " of roughly +-3% in the estimation of uniques with 95% confidence."
+    + " The target type parameter is optional and must be 'HLL_4', 'HLL_6' or 'HLL_8'."
+    + " The default is 'HLL_4'")
+@SuppressWarnings("javadoc")
+public class DataToSketchUDAF2 extends AbstractGenericUDAFResolver {
```

Comment:
I needed to make some minor changes  like instead:
```
Union(data2sketch)
```
there is a  `data2sketch` which is executing union if the input is a struct...now it's name is not 

I now see how I could probably do the same without making this change

I needed some changes to the evaluate function; 

earlier data2sketch and unionsketch internally:
```
data2sketch: [RAW_DATA] -> [INTERNAL_STRUCT_REPR] -> [BYTEA]
unionSketch: [BYTEA] -> [INTERNAL_STRUCT_REPR] -> [BYTEA]
evaluateSketch: [BYTEA] -> [DOUBLE]
```
now:
```
data2sketch: [RAW_DATA] -> [STRUCT_REPR] -> [STRUCT_REPR]
unionSketch: [STRUCT_REPR] -> [STRUCT_REPR] ->[STRUCT_REPR]
evaluateSketch: [STRUCT_REPR] -> [DOUBLE]
```
looks better to me..and if there would be some way to pretty print the sketch; or hide its data...
I think that actually the sketch internally also has the struct field data...

probably it would be great to pack these things into a type or something.